src/main/java/http_server/Session.java

```diff
@@ -26,8 +26,8 @@ public void run() {
             String input = client.readData();
 
             RequestParser parser = new RequestParser(input);
-            Response responder = new Response(parser, router);
-            String response = responder.generateResponse();
+            Request request = new Request(parser.method(), parser.route(), parser.body(), parser.headers());
```

Comment:
For example, the parser and the request itself have some connective tissues - `Session` here is taking the input, creating a parser, and then mapping that parser to arguments in `Request`'s constructor.  If either the parser or request changes, now `Session` has to change as well.

What if there were some intermediary object that handled the composition of these two, though?

```
Request request = RequestBuilder.from(input);
```

Then if something in the parser changes, or the `Request` constructor changes, `Session` might not really need to know about it.

Reply:
To add to this, your existing `RequestParser` class could be responsible for creating a request by calling some method that returns a `Request` object.